halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/job/v1/TeeByteArrayOutputStream.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.halyard.core.job.v1;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class TeeByteArrayOutputStream extends ByteArrayOutputStream {
```

Comment:
I wanted to, but it has no "getBytes" method to read what's been written to it. Without that, I can't provide that as the output stream to the job executor, since (if we ever want to standardize on one job executor) we need to support reading what the job wrote to stdout & stderr at any point.